### PR TITLE
Fix EnumVarP func to show default value associated with EnumVariable …

### PR DESCRIPTION
**Pull request by RamanaReddy0M**
…(#104)

### CODE REVIEW

**Review by tarunKoyalwar — APPROVED** [review]
lgtm!

File: goflags.go [hunk 1]
```diff
@@ -490,7 +490,7 @@ func (flagSet *FlagSet) EnumVarP(field *string, long, short string, defaultValue
 	flagData := &FlagData{
 		usage:        usage,
 		long:         long,
-		defaultValue: defaultValue,
+		defaultValue: *field,
```
[the remark this answers is not in the document]
Comment (RamanaReddy0M) [reply goflags.go:493]: I agree that having a fatal error for invalid enum. But I just wanted to confirm that
``` go
...
	defaultValue: *field,
...
```
instead
``` go
...
flagData := &FlagData{
                 ...
		defaultValue: defaultValue.String(),
	}
...
```
According to #104,  we need the key associated with that EnumVariable.

[the remark this answers is not in the document]
Comment (Mzack9999) [reply goflags.go:493]: just verified I guess later on we will need a rework as `enum.String()` should return the string associated to it

**Review by Mzack9999 — APPROVED** [review]

**Comment by tarunKoyalwar** [comment]
@Mzack9999 , what do you thing about changing default value type and AllowedTypes?

### Current Implementation
```go
flagSet.EnumVarP(&options.ScanStrategy, "scan-strategy", "ss", goflags.EnumVariable(0), "strategy to use while scanning(auto/host-spray/template-spray)", goflags.AllowdTypes{
			"auto":           goflags.EnumVariable(0),
			"host-spray":     goflags.EnumVariable(1),
			"template-spray": goflags.EnumVariable(2),
		})
```
### New Implementation
```go
flagSet.EnumVarP(&options.ScanStrategy, "scan-strategy", "ss", "auto", "strategy to use while scanning(auto/host-spray/template-spray)", []string{"auto","host-spray","template-spray"}
```

we will use `Allowd Types` internally but there doesn't seem a need to expose it and make it complex .
and for validation we can simply do

```go
if _,ok:= AllowedTypes[defaultValue]; !ok{
    panic(fmt.Errof("%v does not belong to %v",defaultValue,AllowedTypes))
}
```

**Comment by Mzack9999** [comment]
@tarunKoyalwar That's what I was discussing with @RamanaReddy0M , I think anyway that a `iota` allows to plug in more generic implementations. Internally it's needed some refactoring as also other types like boolean/ints are considered like strings, losing their original type

**Pull request merged**